### PR TITLE
link clearing capabilities to memoize attributes

### LINKED ISSUE #674: Link __Memoize attribute to HH/clear_static_memoization
The documentation on special internal attributes describes `<<__Memoize>>` and `<<__MemoizeLSB>>`. That might also be a helpful place to mention the existence of https://docs.hhvm.com/hack/reference/function/HH.clear_static_memoization/ and the LSB equivalent because they're useful for testing/working with memoized functions.

### PR DESCRIPTION
**Pull request by AndrewDiMola**
* @ryangreenberg had a great idea to link the HH clearing functions for `__Memoize` and `__MemoizeLSB` on the [attributes page](https://docs.hhvm.com/hack/attributes/predefined-attributes).
* Fixes #674

### CODE REVIEW

File: guides/hack/20-attributes/07-predefined-attributes.md [hunk 1]
```diff
@@ -221,6 +221,8 @@ The types of the parameters are restricted to the following: `null`, `bool`, `in
 
 The interface type `IMemoizeParam` assists with memoizing objects passed to async functions.
 
+To clear the cache and avoid potential memory leaks, use `HH\clear_static_memoization`.
```
Comment (fredemmott) [inline guides/hack/20-attributes/07-predefined-attributes.md:224]: This should not be done in production code; it's really only intended for unit tests, and only if the tests require breaking the isolation of 'a unit'. It shouldn't generally be used.

`<<__Memoize>>` is frequently abused as a 'run at most one time', so clearing caches can really break stuff.

**Review by fredemmott — CHANGES_REQUESTED** [review]
I'm fine with the content here, but it does need some more warnings :)

File: guides/hack/20-attributes/07-predefined-attributes.md (hunk 1 above)
Comment (AndrewDiMola) [reply guides/hack/20-attributes/07-predefined-attributes.md:224]: hmm do we have a way of embedding macro blocks? something like:
image

Comment (fredemmott) [reply guides/hack/20-attributes/07-predefined-attributes.md:224]: if you grep -i src/ for `fbonly` you'll see one special case of this, which could be generalized

**Pull request merged**